doc/architecture/decisions/0005-rooms-and-beds-for-both-cas-services.md

```diff
@@ -0,0 +1,106 @@
+`bed_count` on Premises directly. It was imagined that this would eventually
+change over.
+
+CAS3 are looking to extend mini-manage now in way that works for CAS3 but is
+also compatible for CAS1. Both teams built a shared outstanding going into this
+work:
+
+- Both CAS1 and CAS3 need to represent rooms in some fashion and associate them
+  with bookings in future
+- CAS1 will have rooms with one or multiple beds
+- CAS3 will have one bed per room, it is highly unlikely but still possible
+  that in future this might change
+- Physical spaces such as Premises and Rooms look to have different
+  characteristics for CAS1 and CAS3. For example, CAS3 needs to specify whether
+  a room has a shared kitchen but CAS1 may not, however CAS1 might need to know
+  if it has an en suite.
+
+[1] <https://dbdiagram.io/d/6200fa0285022f4ee54d7569>
+
+## Decision
+
+- We will attempt to build and share the core room and bed resources
+- We will separate out the storage of unique characteristics from
+  rooms and beds in a way that supports future flexibility for both services
+- We imagine[2] we can create a new Characteristics table where records can be
+  created to represent boolean attributes. These could be associated with
+  premises or rooms via two further join tables:
+
+         ,---------------------------------------.
+         |Characteristic                         |
+         |---------------------------------------|
+         |*id: integer <<autonumber>>            |
+         |--                                     |
+         |*name: string                          |
+         |*service_scope: string <<default: "*">>|
+         |*model_scope: string <<default: "*">>  |
+         `---------------------------------------'
+                 |                |
+                 |                |----------`
+                 |                          .|.
+                 |           ,---------------------------.
+                 |           |PremiseCharacteristic      |
+                 |           |---------------------------|
+                 |           |*premise_id: integer       |
+                 |           |*characteristic_id: integer|
+                 |           `---------------------------'
+                 |                           .|.
+                 |                            |
+                 |                            |
+                .|.                  ,---------------------------.
+      ,---------------------------.  |Premise                    |
+      |RoomCharacteristic         |  |---------------------------|
+      |---------------------------|  |*id: integer <<autonumber>>|
+      |*room_id: integer          |  |--                         |
+      |*characteristic_id: integer|  |address: string            |
+      `---------------------------'  |postcode: string           |
+             .|.                     `---------------------------'
```

Comment:
just to nit-pick on the language. Our usage of "premises": 
> a house or building, together with its land and outbuildings ...

is a plural noun -- i.e. "one premises", "two premises" etc

There's another v different usage of the word which does have the singular noun form "premise":

> a previous statement or proposition from which another is inferred or follows as a conclusion

`</ pedantry>`

Reply:
On this - all of our tables names since `20220811100325__pluralize_table_names.sql` are pluralized & snake case so would be good to have these the same (e.g. `room_characteristics`)

Reply:
Absolutely, I'll make those updates.